lib/assert.js

```diff
@@ -400,7 +401,7 @@ function expectedException(actual, expected, msg) {
   if (expected.prototype !== undefined && actual instanceof expected) {
     return true;
   }
-  if (Error.isPrototypeOf(expected)) {
+  if (isPrototypeOf.call(Error, expected)) {
```

Comment:
`Reflect.apply` is preferred over `Function.prototype.call`/`apply`. Ditto everywhere else.

Reply:
I have one concern about using that in `getOrSetAsyncId` since it's somewhat hot path code... means creating an array for the arguments each time.

Reply:
in getOrSetAsyncId shouldn't we really be using an undefined check since it appears to be [a lot faster](https://jsperf.com/hasownproperty-vs-in-vs-undefined/12)

Reply:
@devsnek it still has the problem of potentially inheriting through prototype. I don't know how likely it is though... I'll ping @AndreasMadsen

Reply:
Reflect.has also seems significantly faster at this point, just as another option